dpnp/dpnp_iface_arraycreation.py

```diff
@@ -95,6 +94,37 @@
 ]
 
 
+def _check_limitations(order=None, subok=False, like=None):
+    """
+    Checking limitation kwargs for their supported values.
+
+    Parameter `order` is supported only with values ``"C"``, ``"F"`` and ``None``.
+    Parameter `subok` is supported only with default value ``False``.
+    Parameter `like` is supported only with default value ``None``.
+
+    Raises
+    ------
+    NotImplementedError
+        If any input kwargs is of unsupported value.
+    """
+
+    if order not in ("C", "c", "F", "f", None):
```

Comment:
there might be different exception raised, in case:
```python
dpnp.empty(10, order='S')
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
Cell In[2], line 1
----> 1 dpnp.empty(10, order='S')

File ~/code/dpnp/dpnp/dpnp_iface_arraycreation.py:1121, in empty(shape, dtype, order, like, device, usm_type, sycl_queue)
   1046 def empty(
   1047     shape,
   1048     *,
   (...)
   1054     sycl_queue=None,
   1055 ):
   1056     """
   1057     Return a new array of given shape and type, without initializing entries.
   1058
   (...)
   1118
   1119     """
-> 1121     _check_limitations(order=order, like=like)
   1122     return dpnp_container.empty(
   1123         shape,
   1124         dtype=dtype,
   (...)
   1128         sycl_queue=sycl_queue,
   1129     )

File ~/code/dpnp/dpnp/dpnp_iface_arraycreation.py:112, in _check_limitations(order, subok, like)
     98 """
     99 Checking limitation kwargs for their supported values.
    100
   (...)
    108     If any input kwargs is of unsupported value.
    109 """
    111 if order not in ("C", "c", "F", "f", None):
--> 112     raise NotImplementedError(
    113         "Keyword argument `order` is supported only with "
    114         f"values ``'C'`` and ``'F'``, but got {order}"
    115     )
    116 elif like is not None:
    117     raise NotImplementedError(
    118         "Keyword argument `like` is supported only with "
    119         f"default value ``None``, but got {like}"
    120     )

NotImplementedError: Keyword argument `order` is supported only with values ``'C'`` and ``'F'``, but got S
```
but numpy raises;
```
numpy.empty(10, order='S')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[3], line 1
----> 1 numpy.empty(10, order='S')

ValueError: order must be one of 'C', 'F', 'A', or 'K' (got 'S')
```
I believe we should raise the same exception of type `ValueError` as numpy, when `order` value is out of range for numpy call as well.
Do we have the use case covered by any tests?

Reply:
We have tests that checks unsupported kwargs `like`, `subok`, and `order` for all array creation functions